process/sync/storageBootstrap/baseStorageBootstrapper.go

```diff
@@ -0,0 +1,315 @@
+package storageBootstrap
+
+import (
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/typeConverters"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/logger"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/block/bootstrapStorage"
+	"github.com/ElrondNetwork/elrond-go/process/sync"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+var log = logger.GetOrCreate("process/sync")
+
+// ArgsStorageBootstrapper is structure used to create a new storage bootstrapper
+type ArgsStorageBootstrapper struct {
+	BootStorer          process.BootStorer
+	ForkDetector        process.ForkDetector
+	BlockProcessor      process.BlockProcessor
+	ChainHandler        data.ChainHandler
+	Marshalizer         marshal.Marshalizer
+	Store               dataRetriever.StorageService
+	Uint64Converter     typeConverters.Uint64ByteSliceConverter
+	BootstrapRoundIndex uint64
+	ShardCoordinator    sharding.Coordinator
+	ResolversFinder     dataRetriever.ResolversFinder
+}
+
+type storageBootstrapper struct {
+	bootStorer       process.BootStorer
+	forkDetector     process.ForkDetector
+	blkExecutor      process.BlockProcessor
+	blkc             data.ChainHandler
+	marshalizer      marshal.Marshalizer
+	store            dataRetriever.StorageService
+	uint64Converter  typeConverters.Uint64ByteSliceConverter
+	shardCoordinator sharding.Coordinator
+
+	bootstrapRoundIndex  uint64
+	bootstrapper         storageBootstrapperHandler
+	headerNonceHashStore storage.Storer
+}
+
+func (st *storageBootstrapper) loadBlocks() error {
+	var err error
+	var storageHeaderInfo bootstrapStorage.BootstrapData
+
+	round := st.bootStorer.GetHighestRound()
+	storageHeadersInfo := make([]bootstrapStorage.BootstrapData, 0)
+
+	log.Debug("Load blocks started...")
+
+	for {
+		storageHeaderInfo, err = st.bootStorer.Get(round)
+		if err != nil {
+			break
```

Comment:
Should we treat this? At least a debug messsage

Reply:
this error is treat  outside of **for** instruction